pfsimulator/parflow_lib/reservoir_package.c

```diff
@@ -0,0 +1,533 @@
+      iz_upper = grid_nz - 1;
+
+      nx = 1;
+      ny = 1;
+      nz = 1;
+
+      rx = 0;
+      ry = 0;
+      rz = 0;
+
+      current_mpi_rank = amps_Rank(amps_CommWorld);
+
+      new_intake_subgrid = NewSubgrid(intake_ix, intake_iy, iz_lower,
+                                      nx, ny, nz,
+                                      rx, ry, rz,
+                                      current_mpi_rank);
+
+      dx = SubgridDX(new_intake_subgrid);
+      dy = SubgridDY(new_intake_subgrid);
+      dz = SubgridDZ(new_intake_subgrid);
+
+      intake_subgrid_volume = (nx * dx) * (ny * dy) * (nz * dz);
+
+      new_secondary_intake_subgrid = NewSubgrid(secondary_intake_ix, secondary_intake_iy, iz_lower,
+                                                nx, ny, nz,
+                                                rx, ry, rz,
+                                                current_mpi_rank);
+
+      dx = SubgridDX(new_secondary_intake_subgrid);
+      dy = SubgridDY(new_secondary_intake_subgrid);
+      dz = SubgridDZ(new_secondary_intake_subgrid);
+
+
+      secondary_intake_subgrid_volume = (nx * dx) * (ny * dy) * (nz * dz);
+
+      new_release_subgrid = NewSubgrid(release_ix, release_iy, iz_lower,
+                                       nx, ny, nz,
+                                       rx, ry, rz,
+                                       current_mpi_rank);
+
+      dx = SubgridDX(new_release_subgrid);
+      dy = SubgridDY(new_release_subgrid);
+      dz = SubgridDZ(new_release_subgrid);
+      release_subgrid_volume = (nx * dx) * (ny * dy) * (nz * dz);
+      if (subgrid_lives_on_this_rank(new_intake_subgrid, grid)) {
+        intake_cell_rank = current_mpi_rank;
+      }
+      amps_Invoice intake_cell_invoice = amps_NewInvoice("%i", &intake_cell_rank);
+      amps_AllReduce(amps_CommWorld, intake_cell_invoice, amps_Max);
+      amps_FreeInvoice(intake_cell_invoice);
+      if (subgrid_lives_on_this_rank(new_secondary_intake_subgrid, grid)) {
+        secondary_intake_cell_rank = current_mpi_rank;
+      }
+      amps_Invoice secondary_intake_cell_invoice = amps_NewInvoice("%i", &secondary_intake_cell_rank);
+      amps_AllReduce(amps_CommWorld, secondary_intake_cell_invoice, amps_Max);
+      amps_FreeInvoice(secondary_intake_cell_invoice);
+      if (subgrid_lives_on_this_rank(new_release_subgrid, grid)) {
+        release_cell_rank = current_mpi_rank;
+        release_subgrid_volume = GetSubgridVolume(new_release_subgrid, problem_data);
+      }
+      amps_Invoice release_cell_invoice = amps_NewInvoice("%i", &release_cell_rank);
+      amps_AllReduce(amps_CommWorld, release_cell_invoice, amps_Max);
+      amps_FreeInvoice(release_cell_invoice);
+
+      part_of_reservoir_lives_on_this_rank =
+              (release_cell_rank==current_mpi_rank) ||
+              (intake_cell_rank==current_mpi_rank) ||
+              (secondary_intake_cell_rank==current_mpi_rank);
+
+      MPI_Comm new_reservoir_communicator;
```

Comment:
Will break if MPI is not used.   Need to guard MPI usage.

#ifdef PARFLOW_HAVE_MPI
#else
#endif

The AMPS layer was supposed to hide MPI and do nothing/appropriate when MPI or other parallel layers where not present.   So for things that are not exposed by AMPS we need to guard appropriately or expand the AMPS layer to add the functionality.